### ISSUE TITLE
Support inputting the script in hex bytes

### ISSUE THREAD

**Issue by stevenroose** [dgpv/bsst#1]
Instead of expecting a script in ASM format, it would be nice if I can just have a script be a single-line file with just hex :)

`hal elements script decode` doesn't exist yet and the hal (rust-bitcoin) script assembly creator actually prefixes byte pushes with `OP_PUSHBYTES_n` opcodes.

**Comment by dgpv** [comment]
This would require quite a bit of code to be added. It was an intentional decision that only the assembly will be analyzed, because hex->assembly transformation can be done by other tools. I think better way is to just accomodate more assembly formats. I can just add an option for parser to expect that assembly "prefixes byte pushes with OP_PUSHBYTES_n opcodes".

For more obscure assembly formats, I guess people would just need to convert them to the acceptable format :-) Bitcoin/Elements scripts are not that big for that task to be too laborous :-)

**Comment by stevenroose** [comment]
Yeah, I use `hal script decode`, but I can also quite easily fiddle/tweak that. Even though it's the rust-bitcoin built-in formatting, I can easily implement other formats. I've been thinking about this as well to do for the https://ide.scriptwiz.app/ format.

It's weird that your analysis is on ASM format instead of binary format, though..

**Comment by dgpv** [comment]
> It's weird that your analysis is on ASM format instead of binary format, though..

It allows to support annotations in specially-formatted comments, and to support placeholders (like, `$btcAsset`)

Now annotations are used for marking values on the stack with identifiers, so that report can be easier to read (see https://gist.github.com/dgpv/b57ecf4d9e3d0bfdcc2eb9147c9b9abf -- all these `oracle_tag`, `synth_asset`, etc were given via ` // =>oracle_tag`, ` // =>synth_asset` comments in the source)

In the future bsst might also support annotations that assert certain conditions, like, "in this place, value on the stack must be equal 42" -- without having to put extra code `DUP 42 NUMEQUALVERIFY` into the script. This can help to ease the solving, because now a check for fairly large scripts can take a long time. Still reasonable time, but if practical scripts start to get larger, we might need these intermediate assertions to help the solver.

**Comment by dgpv** [comment]
> Even though it's the rust-bitcoin built-in formatting, I can easily implement other formats.

I think this would be the best option.

It is possible to add a feature to decode scripts with `python-bitcointx`, `python-elementstx`, but that would basically mean adding and maintaining an extra tool - the script decoder. I would prefer to avoid that.

**Comment by dgpv** [comment]
After #22 is merged, a plugin that installs `parse_input_file` hook will be able to process the input file in any desired way, returning `ScriptInfo` for further analysis
